components/prism-naniscript.js

```diff
@@ -0,0 +1,189 @@
+var expressionDef = function (isMultiline) {
+	if (isMultiline === void 0) { isMultiline = true; }
+	return isMultiline
+		? /\{[^\r\n\[\]{]*\}(?=[^\r\n\[\]\}]*)/m
+		: /\{[^\r\n\[\]{}]*\}/;
+};
+
+Prism.languages.naniscript = {
+	// ; ...
+	'comment': {
+		pattern: /^([\t ]*);[^\r\n]*$/m,
+		lookbehind: true,
+	},
+	// > ...
+	// Define is a control line starting with '>' followed by a word, a space and a text.
+	'define': {
+		pattern: /^>.+$/m,
+		alias: 'tag',
+		inside: {
+			value: {
+				pattern: /(^>[a-zA-Z0-9_]+)[\t ]+[^{}\r\n]+/,
+				alias: 'operator',
+				lookbehind: true,
+			},
+			key: {
+				pattern: /(^>)[a-zA-Z0-9_]+/,
+				lookbehind: true,
+			}
+		}
+	},
+	// # ...
+	'label': {
+		pattern: /^([\t ]*)[#]{1}[\t ]*[a-zA-Z0-9]+[\t ]?$/m,
+		alias: 'regex',
+		lookbehind: true
+	},
+	'command': {
+		pattern: /^([\t ]*)@[a-zA-Z0-9_]+(?=[\t ]+[\S]{1,}|[\t ]+['"][\S]|[\t ]+$|$).*/m,
+		lookbehind: true,
+		alias: 'function',
+		inside: {
+			expression: {
+				pattern: expressionDef(),
+				greedy: true,
+				alias: 'selector'
+			},
+			'whitespaces': /[\t ]+$/,
+			'command-name': /^[\t ]*@[a-zA-Z0-9_]+/,
+			'command-params': {
+				pattern: /[^\n]+$/,
+				inside: {
+					'quoted-string': {
+						pattern: /"(?:[^"\\]|\\.)*"/,
+						alias: 'operator'
+					},
+					'command-param-id': {
+						alias: 'property',
+						pattern: /([^"\\]?) [a-zA-Z0-9_]+:/,
+						lookbehind: true,
+					},
+					'command-param-value': [
+						{
+							pattern: expressionDef(false),
+							alias: 'selector',
+						},
+						{
+							pattern: /([\t ]+).+?(?=[\t ]|$)/,
+							alias: 'operator',
+							greedy: true,
+							lookbehind: true
+						},
+						{
+							pattern: /.+/,
+							alias: 'operator',
+						}
+					]
+				}
+			},
+		}
+	},
+	// Generic is any line that doesn't start with operators: ;>#@
+	'generic-text': {
+		pattern: /^[^#@>;\r\n]{1}.*$/m,
+		alias: 'punctuation',
+		inside: {
+			// \{ ... \} ... \[ ... \] ... \"
+			'escaped-char': {
+				pattern: /\\[{}\[\]"]{1}/m
+			},
+			expression: {
+				pattern: expressionDef(),
+				greedy: true,
+				alias: 'selector'
+			},
+			'inline-command': {
+				pattern: /\[[\t ]*[a-zA-Z0-9_]+[^\r\n\[]*\]/m,
+				greedy: true,
+				alias: 'function',
+				inside: {
+					'quoted-string': {
+						pattern: /"(?:[^"\\]|\\.)*"/
+					},
+					'command-param-id': {
+						alias: 'property',
+						pattern: /([^"\\]?)[a-zA-Z0-9_]+:/,
+						lookbehind: true,
+					},
+					'command-param-name': {
+						pattern: /^(\[)[\t ]*[a-zA-Z0-9_]+/,
+						alias: 'name',
+						lookbehind: true
+					},
+					'start-stop-char': /[\[\]]{1}/,
+					'command-param-value': [
+						{
+							pattern: expressionDef(false),
+							alias: 'selector',
+						},
+						{
+							pattern: /([\t ]+).+?(?=[\t ]|$)/,
+							alias: 'operator',
+							lookbehind: true
+						},
+						{
+							pattern: /.+/,
```

Comment:
This also captures all spaces between parameters.
```suggestion
							pattern: /\S(?:.*\S)?/,
```